service/matching/matchingEngine.go

```diff
@@ -745,6 +745,59 @@ func (e *matchingEngineImpl) GetWorkerBuildIdOrdering(
 	}, nil
 }
 
+func (e *matchingEngineImpl) InvalidateTaskQueueMetadata(
+	hCtx *handlerContext,
+	req *matchingservice.InvalidateTaskQueueMetadataRequest,
+) (*matchingservice.InvalidateTaskQueueMetadataResponse, error) {
+	namespaceID := namespace.ID(req.GetNamespaceId())
+	taskQueueName := req.GetTaskQueue()
+	taskQueue, err := newTaskQueueID(namespaceID, taskQueueName, enumspb.TASK_QUEUE_TYPE_WORKFLOW)
+	if err != nil {
+		return nil, err
+	}
+	tqMgr, err := e.getTaskQueueManager(hCtx, taskQueue, enumspb.TASK_QUEUE_KIND_NORMAL, true)
+	if err != nil {
+		if _, ok := err.(*serviceerror.NotFound); ok {
```

Comment:
Hmm. Looks like it doesn't... so not sure where I got this idea. It _does_ look like it can return `nil, nil` though, so changed it to deal with that.